(SERVER-1266) Import developer dashboard from PE

camlow325 opened the pull request:
This PR imports the code for the Puppet Server Developer dashboard from
PE Puppet Server.  This PR is based on work in #1335.  Only the last 2 commits
are unique for this PR.  I'll rebase this PR once the related PR is merged but
wanted to put this up for initial review.

rlinehan commented on src/cljs/puppetlabs/puppetserver/dashboard/dashboard/production.cljs:
```diff
@@ -0,0 +1,4 @@
+(ns puppetlabs.puppetserver.dashboard.production
```
I think this file path is wrong - it has an extra level of `dashboard`

camlow325 replied:
Thanks @rlinehan.  I bumped this and the other cljs files up a level in e698ecb since this appears to more closely match what we did in PE Puppet Server.

camlow325 commented:
I added an auth.conf rule and associated beaker test in 63dfb2f.

When testing this with an ezbake build of puppetserver, I found that adding the figwheel plugin had broken the ability to use the ezbake plugin locally.  I added an exclusion in the figwheel plugin in c7b4b39.  With that change in place, I confirmed that `lein figwheel` worked as well as it did for me on the previous commit.  It at least hosted the top-level html and js files on port 3449.  In order to get most of the metrics widgets from the cljs-dashboard-widgets project to load in the figwheel server, I had to manually clone the cljs-dashboard-widgets repo locally and run:

```
ln -s <cljs-dashboard-widgets-repo-clone>/resources/puppetlabs/metrics/dashboard/public <puppetserver-repo-clone>/target/js-resources/puppetlabs/puppetserver/public/metrics-dashboard
```

Attempts by the js to query the status endpoint failed because the status endpoint apparently isn't being hosted on port 3449.  I'm not really sure how to address that issue.

rlinehan commented:
@camlow325 sorry, I think this needs to be rebased now.

jpinsonault commented:
I'm 👍

camlow325 commented:
Rebased at edb9592.

rlinehan commented:
💯
